src/ui/components/BalanceInput.tsx

```diff
@@ -0,0 +1,60 @@
+import React, {useEffect, useState} from 'react';
+import {StyleSheet} from 'react-native';
+import {TextInput, useTheme} from '@ui/library';
+import {useFormatBalance} from 'src/api/hooks/useFormatBalance';
+import {decimalKeypad} from 'src/utils';
+import MaxBalance from './MaxBalance';
+import {getBalanceFromString} from 'src/api/utils/balance';
+import {ApiPromise} from '@polkadot/api';
+import type {Account} from 'src/api/hooks/useAccount';
+
+type PropTypes = {
+  api?: ApiPromise;
```

Comment:
@ishrak-imam  `API` is still optional, will have to call useApi() in the component?

Reply:
No, let's just not block the rendering of the BalanceInput component.

Reply:
I think this issue should be blocked by https://github.com/litentry/litentry-app/issues/956. Or at least have another issue to refactor this and it's usages afterwards

Reply:
@franjs @Ekaanth 
I think we can let this PR go through after the review comments has been taken care of. It will make the refactor of #956 easier since most of the usage usage of the `useFormatBalance` and `getBalanceFromString` are isolated in the `BalanceInput` component. Let's not block it by #946 

And we can create a followup ticket for the keep alive functionality.

Reply:
+1